src/zir_sema.zig

```diff
@@ -488,6 +489,25 @@ fn analyzeInstCompileError(mod: *Module, scope: *Scope, inst: *zir.Inst.CompileE
     return mod.fail(scope, inst.base.src, "{}", .{inst.positionals.msg});
 }
 
+fn analyzeInstCompileLog(mod: *Module, scope: *Scope, inst: *zir.Inst.CompileLog) InnerError!*Inst {
+    std.debug.print("| ", .{});
+    for (inst.positionals.to_log) |item, i| {
+        const to_log = try resolveInst(mod, scope, item);
+        if (to_log.value()) |val| {
+            std.debug.print("{}", .{val});
+        } else {
+            std.debug.print("(runtime value)", .{});
+        }
+        if (i != inst.positionals.to_log.len - 1) std.debug.print(", ", .{});
+    }
+    std.debug.print("\n", .{});
+    switch (mod.fail(scope, inst.base.src, "found compile log statement", .{})) {
```

Comment:
The compiler cannot currently handle one decl failing multiple times and this will lead to an assertion failure in `Module.failWithOwnedErrorMsg`.

Reply:
Ah. I thought this may happen. Stage1 adds a check bool in the inst that is set to true if the compileLog is already referenced. Is that the best way to do it? The relevant stage1 source is ir.cpp:24558.

Reply:
That check is to avoid adding an error every time the compilelog is analyzed which is only relevant in comptime function calls. Eg. the following only adds one compile error.
```zig
fn aa(comptime a: u32) void {
    @compileLog(a);
}

comptime {
    aa(1);
    aa(2);
}
```

Stage2 will eventually need this flag too but for now the issue is that `failed_decls` only takes one error message per failed decl.

Reply:
Interesting, stage1 can have multiple compiler errors on a decl. IMO, its a bug in stage2, because if a mod.fail is issued for a compileLog, a real compile error will trigger the assertion. I got it to work with two compile logs by checking if the decl is in `mod.failed_decls`, but I am convinced that this is a bad/temporary solution because this triggers the assertion still (x(x) causes mod.fail):
```zig
export fn _start() noreturn {
@compileLog(x);
// @compileLog(x);
x(x);
exit();
}
fn x() void {}
```
Maybe a better idea would be to print the compile error to stderr and not have it go through the mod.fail api? Or should we try to restructure the fail api so that a decl can have multiple errors like in stage1? What do you think?